cli/internal/cmd/apply.go

```diff
@@ -40,6 +40,82 @@ import (
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
+const (
+	// skipInfrastructurePhase skips the Terraform apply of the apply process.
+	skipInfrastructurePhase skipPhase = "infrastructure"
+	// skipInitPhase skips the init RPC of the apply process.
+	skipInitPhase skipPhase = "init"
+	// skipAttestationConfigPhase skips the attestation config upgrade of the apply process.
+	skipAttestationConfigPhase skipPhase = "attestationconfig"
+	// skipCertSANsPhase skips the cert SANs upgrade of the apply process.
+	skipCertSANsPhase skipPhase = "certsans"
+	// skipHelmPhase skips the helm upgrade of the apply process.
+	skipHelmPhase skipPhase = "helm"
+	// skipImagePhase skips the image upgrade of the apply process.
+	skipImagePhase skipPhase = "image"
+	// skipK8sPhase skips the Kubernetes version upgrade of the apply process.
+	skipK8sPhase skipPhase = "k8s"
+)
+
+// skipPhase is a phase of the upgrade process that can be skipped.
+type skipPhase string
+
+// skipPhases is a list of phases that can be skipped during the upgrade process.
+type skipPhases map[skipPhase]struct{}
+
+// contains returns true if the list of phases contains the given phase.
+func (s skipPhases) contains(phase skipPhase) bool {
+	_, ok := s[skipPhase(strings.ToLower(string(phase)))]
+	return ok
+}
+
+// add a phase to the list of phases.
+func (s *skipPhases) add(phases ...skipPhase) {
+	if *s == nil {
+		*s = make(skipPhases)
+	}
+	for _, phase := range phases {
+		(*s)[skipPhase(strings.ToLower(string(phase)))] = struct{}{}
+	}
+}
+
+func printAllPhases() string {
```

Comment:
nit/optional: to me, the name implies that the func prints to stdout or the like. Maybe there's a better name.